### PR TITLE
feat(base-driver): Add a route for new window creation

### PR DESCRIPTION
**Pull request by mykola-mokhnach**
## Proposed changes

https://w3c.github.io/webdriver/#new-window

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

cc @titusfortner

### CODE REVIEW

**Comment by mykola-mokhnach** [comment]
Linking https://github.com/appium/appium/issues/16749

**Review by KazuCocoa — APPROVED** [review]

File: packages/types/lib/index.ts [hunk 1]
```diff
@@ -405,6 +406,11 @@ export interface Rect {
   height: number;
 }
 
+export interface NewWindow {
```
Comment (boneskull) [inline packages/types/lib/index.ts:409]: is this type unique to Appium or is it standardized somewhere? (could we borrow the type from wdio?)

File: packages/types/lib/index.ts [hunk 2]
```diff
@@ -405,6 +406,11 @@ export interface Rect {
...
+  handle: string;
+  type: string;
```
Comment (boneskull) [inline packages/types/lib/index.ts:411]: could `type` be anything, or is it only a handful of different strings?  if the latter, list them in a union:

```ts
  type: 'foo' | 'bar' | 'baz'
```

**Review by boneskull — APPROVED** [review]
lgtm, just a couple questions.

I'm curious if TS warned you that `createNewWindow` must exist in the `ExternalDriver` type?

File: packages/types/lib/index.ts (hunk 1 above)
Comment (mykola-mokhnach) [reply packages/types/lib/index.ts:409]: not sure how/where to check this

**Comment by mykola-mokhnach** [comment]
> lgtm, just a couple questions.
> 
> I'm curious if TS warned you that `createNewWindow` must exist in the `ExternalDriver` type?

yes, the linter was complaining to a missing type declaration after only the route itself has been added

File: packages/types/lib/index.ts (hunk 1 above)
Comment (boneskull) [reply packages/types/lib/index.ts:411]: I'm not sure either.  I was just grepping around.  it doesn't look like this response is typed.

File: packages/types/lib/index.ts [hunk 3]
```diff
@@ -406,9 +406,11 @@ export interface Rect {
   height: number;
 }
 
+export type NewWindowType = 'tab' | 'window';
```
Comment (boneskull) [inline packages/types/lib/index.ts:409]: nice, thanks

**Pull request merged**